frontend/src/component/project/Project/ProjectSettings/ProjectActions/ProjectActionsTable/ProjectActionsModal/ProjectActionsForm/ActionItem.tsx

```diff
@@ -0,0 +1,134 @@
+import { IconButton, Tooltip, styled } from '@mui/material';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { IAction } from 'interfaces/action';
+import { Fragment } from 'react';
+import GeneralSelect, {} from 'component/common/GeneralSelect/GeneralSelect';
+import { Delete } from '@mui/icons-material';
+import { useProjectEnvironments } from 'hooks/api/getters/useProjectEnvironments/useProjectEnvironments';
+import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
+import mapValues from 'lodash.mapvalues';
+import { useFeatureSearch } from 'hooks/api/getters/useFeatureSearch/useFeatureSearch';
+import {
+    BoxSeparator,
+    Col,
+    InnerBoxHeader,
+    Row,
+    StyledInnerBox,
+} from './InnerContainerBox';
+
+export type UIAction = Omit<IAction, 'id' | 'createdAt' | 'createdByUserId'> & {
+    id: string;
+};
+
+export const Action = ({
```

Comment:
Something like this would be more consistent with our code base:
```suggestion
interface IProjectActionsFormActionItemProps {
    action: UIAction;
    index: number;
    stateChanged: (action: UIAction) => void;
    onDelete: () => void;
}

export const ProjectActionsFormActionItem = ({
    action,
    index,
    stateChanged,
    onDelete,
}: IProjectActionsFormActionItemProps) => {
```
I also recommend renaming the component as `ActionItem` is a bit generic.